src/audio/audio.cpp

```diff
@@ -486,16 +498,30 @@ void Audio::doCapture()
     }
 #endif
 
-    for (size_t i = 0; i < AUDIO_FRAME_SAMPLE_COUNT * AUDIO_CHANNELS; ++i)
-        buf[i] *= inGain;
+    for (quint32 i = 0; i < AUDIO_FRAME_SAMPLE_COUNT * AUDIO_CHANNELS; ++i)
+    {
+        int16_t ampPCM = static_cast<int16_t>(qRound(buf[i] * gain));
+        buf[i] = static_cast<int16_t>(ampPCM);
+    }
 
     emit frameAvailable(buf, AUDIO_FRAME_SAMPLE_COUNT, AUDIO_CHANNELS, AUDIO_SAMPLE_RATE);
 }
 
 /**
+@internal
+*/
+void Audio::privateSetGain(qreal percentage)
+{
+    percentage = std::max(0.0, std::min(percentage, 1.0));
+    gainPercentage = percentage;
+
+    gain = (percentage > 0.0) ? 10.0 * qAbs(log10(percentage * 100.0)) : 0.0;
```

Comment:
Yeah, I just remembered why I put it there. Actually removing the abs changes "nothing" here as values are always within range 1..100 (percentage * 100.0) -> I'll re-add the abs to be on the safe side. Btw. Percentage is already clipped above.

Still 1% results in a 0 gain factor…??